src/main/java/mate/project/service/ShoppingCartServiceImpl.java

```diff
@@ -92,9 +92,8 @@ private User getAuthenticatedUser() {
     }
 
     private ShoppingCart getUsersShoppingCart(User user) {
-        return shoppingCartRepository.findByUser(user)
+        return shoppingCartRepository.findByUserWithCartItems(user)
```

Comment:
The method name `findByUserWithCartItems` suggests that it retrieves a user along with their cart items. Ensure that this method is implemented correctly in the repository and that it returns the expected data structure. If this method is new, verify its functionality with appropriate tests.